src/loaders/compute_tools/tool_common.py

```diff
@@ -252,11 +260,17 @@ def run_single(self, tool_callable: Callable[[str, Path, Path, bool], None]):
         self._execute(self._threads, tool_callable, args_list, start, False)
         _create_metadata_file(genomes_meta, batch_dir)
     
-    def run_batched(self, tool_callable: Callable[[Dict[str, Path], Path, int, bool], None]):
+    def parallel_batch_execution(self, tool_callable: Callable[[Dict[str, Path], Path, int, bool], None]):
         """
         Run a tool in batched mode, where > 1 data file is processed by the tool in one
         call. Each batch gets its own batch directory.
 
+        Data IDs are divided into batches, and each batch is processed in parallel. The tool execution results can
+        be consolidated into individual files for each batch
+
+        Use case: checkm2 - concurrently execute checkm2 on a batch of genomes, and one result file (quality_report.tsv)
```

Comment:
I think the canonical use case here is gtdb_tk, where batching input means much faster overall throughput

Reply:
👍